### DISCUSSION TITLE
[Question] how to pass the output of task-1 as input of task-2?

### DISCUSSION THREAD

**Discussion by mahone3297** [crewAIInc/crewAI#1154] [Q&A]
In the `readme.md`, there is an example:

```python
researcher = Agent()
writer = Agent()

# Create tasks for your agents
task1 = Task()

task2 = Task()

# Instantiate your crew with a sequential process
crew = Crew(
  agents=[researcher, writer],
  tasks=[task1, task2],
  verbose=2, # You can set it to 1 or 2 for different logging levels
  process=Process.sequential
)

# Get your crew to work!
result = crew.kickoff()
```

How can the output of `task-1` be used as the input for `task-2`?
- Can it only be done by modifying the task's prompt and letting the crew AI collaborate on its own? Can't it be controlled by code?

**Comment by matt-crewai** [comment]
You can try  adding the context attribute to Task-2

```
task2 = Task(
    description="Write an article summarising the research findings on AI advancements",
    agent=writer,
    expected_output="A brief article summarising the research findings on AI advancements in Markdown format
    context=[task1]  # This tells CrewAI to use the output of task1 as context for task2
)
```

**Comment by mahone3297** [comment]
> You can try adding the context attribute to Task-2
> 
> ```
> task2 = Task(
>     description="Write an article summarising the research findings on AI advancements",
>     agent=writer,
>     expected_output="A brief article summarising the research findings on AI advancements in Markdown format
>     context=[task1]  # This tells CrewAI to use the output of task1 as context for task2
> )
> ```

i'll try, thx.

**Comment by mdhuzaifapatel** [comment]
> You can try adding the context attribute to Task-2
> 
> ```
> task2 = Task(
>     description="Write an article summarising the research findings on AI advancements",
>     agent=writer,
>     expected_output="A brief article summarising the research findings on AI advancements in Markdown format
>     context=[task1]  # This tells CrewAI to use the output of task1 as context for task2
> )
> ```

I tried this, but still it's not using the output of task1. It worked only once and does not work every other time I tried. Please let me know what can be done?

**Comment by mahone3297** [comment]
When I went to verify, it worked without adding context. I tried several times and it worked every time. I'm not sure why. 
Maybe need to look at the source code.
